jpos/src/main/java/org/jpos/util/DailyLogListener.java

```diff
@@ -148,6 +158,38 @@ public synchronized  void logRotate() throws IOException {
         openLogFile();
         compress(dest);
     }
+
+	public synchronized void deleteOldLogs() throws IOException {
+
+		if (maxAge <= 0) {
+			logDebug("maxage feature is disabled.");
+			return;
+		}
+
+		Path prefixPath = Paths.get(prefix);
+		Path logNamePath = prefixPath.subpath(prefixPath.getNameCount() - 1, prefixPath.getNameCount());
+
+		String rotatedLogRegex = ".*" + logNamePath.toString() + ".+\\" + suffix + "\\" + compressedSuffix;
```

Comment:
@gregorioosorio I don't understand the rationale of the ".*", why don't you just match the basename part  of the files in the log directory? this could delete filles that don't start with prefix name. 
I would keep the basename part, and then in the lambda expression check the basename part against the regex instead of the full path, I don't know if I explain myself here

Reply:
@alcarraz with the **".*"** I tried to support the operating system path separator. I agree with you, is simpler to check only with basename part of the file, instead of the whole path, in fact, I will include the characters ^ and $ in the default regex to make sure that match only a file with the exact name. Finally, I will take in count your comment about to include the parameter: _maxage-delete-regex_, which will have the default regex.
I will push the changes shortly! If there is something I am missing, let me know!

Reply:
I think that would be perfect @gregorioosorio, thank you very much for taking the time to improve this. I was going to take that issue some day, but I don't think I could have done it before next year.